crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1425,9 +1425,23 @@ impl<'db> TypeInferenceBuilder<'db> {
         };
         let value_type = self.infer_expression(value);
 
-        // TODO(charlie): Add remaining branches for different types of augmented assignments.
-        if let (Operator::Sub, Type::Instance(class)) = (*op, target_type) {
-            let class_member = class.class_member(self.db, "__isub__");
+        if let Type::Instance(class) = target_type {
+            let dunder_method = match op {
+                Operator::Add => "__iadd__",
+                Operator::Sub => "__isub__",
+                Operator::Mult => "__imul__",
+                Operator::MatMult => "__imatmul__",
+                Operator::Div => "__itruediv__",
+                Operator::Mod => "__imod__",
+                Operator::Pow => "__ipow__",
+                Operator::LShift => "__ilshift__",
+                Operator::RShift => "__irshift__",
+                Operator::BitOr => "__ior__",
+                Operator::BitXor => "__ixor__",
+                Operator::BitAnd => "__iand__",
+                Operator::FloorDiv => "__ifloordiv__",
+            };
```

Comment:
I would probably add another method to the `Operator` enum itself, similar to the existing ones; this could be generally useful for other users of the AST as well:

https://github.com/astral-sh/ruff/blob/1607d88c2253473a116155176effa4ab14eafcce/crates/ruff_python_ast/src/nodes.rs#L2975-L3008